packages/react-ui-core/src/Listing/ListingSchemas.js

```diff
@@ -0,0 +1,110 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+
+const ApartmentComplexSchema = ({ children, ...rest }) => (
+  <div itemType="https://schema.org/ApartmentComplex" itemScope {...rest}>
+    {children}
+  </div>
+)
+
+ApartmentComplexSchema.propTypes = {
+  children: PropTypes.node,
+}
+
+const AggregateRatingSchema = ({ score, count, reviews }) => {
+  if (!count && !reviews) {
+    return null
+  }
+
+  return (
+    <div
+      itemScope
+      itemProp="aggregateRating"
+      itemType="https://schema.org/AggregateRating"
+    >
+      <meta itemProp="worstRating" content="1.0" />
+      <meta itemProp="bestRating" content="5.0" />
+      <meta itemProp="ratingValue" content={score} />
+      {!!count && <meta itemProp="ratingCount" content={count} />}
+      {!!reviews && <meta itemProp="reviewCount" content={reviews} />}
+    </div>
+  )
+}
+
+AggregateRatingSchema.propTypes = {
+  score: PropTypes.number,
+  count: PropTypes.number,
+  reviews: PropTypes.number,
+}
+
+const AddressSchema = ({ addressLine1, city, state, zip }) => (
+  <span itemType="https://schema.org/PostalAddress" itemProp="address" itemScope>
+    <meta itemProp="streetAddress" content={addressLine1} />
+    <meta itemProp="addressLocality" content={city} />
+    <meta itemProp="addressRegion" content={state} />
+    <meta itemProp="postalCode" content={zip} />
+  </span>
+)
+
+AddressSchema.propTypes = {
+  addressLine1: PropTypes.string,
+  city: PropTypes.string,
+  state: PropTypes.string,
+  zip: PropTypes.string,
+}
+
+const GeoSchema = ({ latitude, longitude }) => {
+  if (!latitude || !longitude) {
+    return null
+  }
+
+  return (
+    <div itemProp="geo" itemScope itemType="https://schema.org/GeoCoordinates">
+      <meta itemProp="latitude" content={latitude} />
+      <meta itemProp="longitude" content={longitude} />
+    </div>
+  )
+}
+
+GeoSchema.propTypes = {
+  latitude: PropTypes.number,
+  longitude: PropTypes.number,
+}
+
+@themed(['PropertyLink'], { pure: true })
+class NameAndUrlSchema extends PureComponent {
+  render() {
+    const { url, theme, children } = this.props
+    return (
+      <div itemProp="name">
+        {url ?
+          <a
+            href={url}
+            itemProp="url"
+            target="_blank"
+            className={theme.PropertyLink}
+            onClick={event => event.preventDefault()}
+          >
+            { children }
+          </a>
+          : children
+        }
+      </div>
+    )
+  }
+}
+
+NameAndUrlSchema.propTypes = {
+  url: PropTypes.string,
+  theme: PropTypes.object,
+  children: PropTypes.node,
+}
+
+export default {
+  AddressSchema,
+  AggregateRatingSchema,
+  ApartmentComplexSchema,
+  GeoSchema,
+  NameAndUrlSchema,
```

Comment:
You could remove "Schema" from all of the schemas, since they're all "namespaced" under `ListingSchemas`.

Reply:
Yeah this is a tough one, I think. If they were imported directly as like an "ApartmentComplex" it might be pretty unclear what that component is trying to do.

Reply:
I think if we create a root component dir that was `Schema` and exported them all from there, maybe then you could remove the suffix? I kind of agree the name is a little unnecessary and if you know that `import { ApartmentComplex } from '../Schema'` I think it's safe to assume where it came from 😄 .